runtime/common/RecordLogParser.h

```diff
@@ -105,6 +105,16 @@ class BufferHandler {
   template <typename T>
   size_t allocateArrayRecord(size_t arrSize) {
     size_t vectorOffset = buffer.size();
+    if constexpr (std::is_same_v<T, bool>) {
+      auto *allocation = new std::vector<bool>(arrSize);
+      if (!allocation)
+        throw std::runtime_error("Memory allocation failed");
```

Comment:
Just use malloc here to avoid the C++ overhead.
```c++
  result = malloc(sizeof(std::vector<bool>) * arrSize);
  if (!result) ...
```
Specifically, `new` will throw exceptions.

Reply:
Wouldn’t it not be exact memory allocate though? Size(vecvor<bool>) is a constant (40 on my machine) and I am not sure how much it will actually use for the elements and where… so won’t we introduce potential memory leaks or bad memory access this way?

Reply:
In short, no.

Reply:
resolved offline